front/app/components/FormBuilder/components/FormFields/FormField/index.tsx

```diff
@@ -60,52 +88,278 @@ export const FormField = ({
 
   const isFieldGrouping = ['page', 'section'].includes(field.input_type);
 
+  // Group is only deletable when we have more than one group
+  const isGroupDeletable =
+    formCustomFields.filter((field) => field.input_type === groupingType)
+      .length > 1;
+  const isDeleteShown =
+    !(field?.input_type !== groupingType || isGroupDeletable) ||
+    get(lockedAttributes, 'enabled', false);
+
   const editFieldAndValidate = () => {
     onEditField({ ...field, index });
     trigger();
   };
 
+  function addCopyIndicatorToTitle(title_multiloc) {
+    const copiedTitle_multiloc = {};
+    Object.keys(title_multiloc).forEach((lang) => {
+      const originalTitle = title_multiloc[lang];
+      copiedTitle_multiloc[lang] =
+        originalTitle !== ''
+          ? `${originalTitle} (${formatMessage(messages.copyNoun)})`
+          : '';
+    });
+    return copiedTitle_multiloc;
+  }
+
+  function duplicateField(originalField: IFlatCustomField) {
+    const {
+      id: _id,
```

Comment:
Not clear why we're using underscores here? Is it to differentiate between the original and the duplicate field?